autolens/point/solver.py

```diff
@@ -302,3 +322,28 @@ def steps(
             )
 
             initial_triangles = up_sampled
+
+    def tree_flatten(self):
+        return (), (
+            self.scale,
+            self.y_min,
+            self.y_max,
+            self.x_min,
+            self.x_max,
+            self.pixel_scale_precision,
+            self.magnification_threshold,
+            self.array_triangles_cls,
```

Comment:
I think because none of the constructor arguments of the solver are derived from search parameters that means that they are static and count as aux data (e.g. they won't change and so don't need to be accounted for in the gradient)